src/main/java/teammates/ui/newcontroller/StudentGetCourseDetailsAction.java

```diff
@@ -0,0 +1,176 @@
+            return new JsonResult(String.format("Course with the given course id (%s) cannot be found.",
+                    courseId), HttpStatus.SC_NOT_FOUND);
+        }
+        gateKeeper.verifyAccessible(logic.getStudentForGoogleId(courseId, userInfo.id), course);
+
+        StudentAttributes student = logic.getStudentForGoogleId(courseId, userInfo.id);
+        if (student == null) {
+            return new JsonResult(String.format("Student with id (%s) cannot be found.", userInfo.id),
+                    HttpStatus.SC_NOT_FOUND);
+        }
+        student.lastName = null;
+        student.comments = null;
+        student.key = null;
+
+        List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+        List<InstructorDetails> instructorDetailss = new LinkedList<>();
+        if (instructors != null) {
+            instructors.forEach(
+                    instructor -> instructorDetailss.add(
+                            new InstructorDetails(instructor.getName(), instructor.getEmail())));
+        }
+
+        TeamDetailsBundle teamDetails = logic.getTeamDetailsForStudent(student);
+        if (teamDetails == null) {
+            return new JsonResult(String.format("Team for the student with id (%s) cannot be found.", userInfo.id),
+                    HttpStatus.SC_NOT_FOUND);
+        }
+
+        List<StudentProfileAttributes> teammateProfiles = new LinkedList<>();
+        if (teamDetails.students != null) {
+            teamDetails.students.removeIf(st -> st.googleId.equals(userInfo.id));
+
+            teamDetails.students.forEach(teammate -> {
+                StudentProfileAttributes teammateProfile = logic.getStudentProfile(teammate.googleId);
+                if (teammateProfile != null) {
+                    teammateProfile.googleId = null;
+                    teammateProfile.modifiedDate = null;
+
+                    teammateProfiles.add(teammateProfile);
+                }
+            });
+        }
+
+        StudentGetCourseDetailsResult result = new StudentGetCourseDetailsResult(
+                student, course, instructorDetailss, teammateProfiles);
+
+        return new JsonResult(result);
+    }
+
+    private boolean hasStudentJoinedCourse(String courseId) {
+        if (userInfo != null && userInfo.isStudent) {
+            return logic.getStudentForGoogleId(courseId, userInfo.id) != null;
+        }
+        return false;
+    }
+
+    /**
+     * A data model to contain details of an instructor.
+     */
+    public static class InstructorDetails {
+
+        private final String name;
+        private final String email;
+
+        public InstructorDetails(String name, String email) {
+            this.name = name;
+            this.email = email;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getEmail() {
+            return email;
+        }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
Do you need `equals` and `hashCode`?

Reply:
i need the equals for comparing actual and expected results in BE test. The hashcode function is needed due to static analysis (PMD) requirement.